### ISSUE TITLE
Doxygen doesnt generated when function has attribute specifier

### ISSUE THREAD

**Issue by Disservin** [microsoft/vscode-cpptools#10995]
### Environment

Version: 1.78.2 (user setup)
Commit: b3e4e68a0bc097f0ae7907b217c1119af9e03435
Date: 2023-05-10T14:39:26.248Z
Electron: 22.5.2
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: No

- C/C++ Extension Version: v1.15.4

### Bug Summary and Steps to Reproduce

Bug Summary:
Doxygen is not auto generated when function has a attribute specifier.

Steps to reproduce:
```cpp
[[nodiscard]] int f() { return 1; }

int main(int argc, char const *argv[]) {
  int x = f();
  return 0;
}
```

1. Start typing `///` above `f` and nothing happens. Works just fine for main..

Expected behavior:
```cpp
/// @brief 
/// @return 
[[nodiscard]] int f() { return 1; }

/// @brief 
/// @param argc 
/// @param argv 
/// @return 
int main(int argc, char const *argv[]) {
  f();
  return 0;
}
```

### Configuration and Logs

```shell
LSP: textDocument/didChange: file:///users/user/Downloads/test.cpp
LSP: cpptools/textEditorSelectionChange
LSP: textDocument/completion: file:///users/user/Downloads/test.cpp (id: 738)
auto_complete::handle_completion: file:///users/user/Downloads/test.cpp (1:2)
LSP: cpptools/getFoldingRanges: file:///users/user/Downloads/test.cpp (id: 739)
LSP: textDocument/didChange: file:///users/user/Downloads/test.cpp
LSP: cpptools/textEditorSelectionChange
LSP: textDocument/completion: file:///users/user/Downloads/test.cpp (id: 740)
auto_complete::handle_completion: file:///users/user/Downloads/test.cpp (1:3)
LSP: cpptools/getFoldingRanges: file:///users/user/Downloads/test.cpp (id: 741)
LSP: cpptools/getCodeActions: file:///users/user/Downloads/test.cpp (id: 742)
LSP: cpptools/getSemanticTokens: file:///users/user/Downloads/test.cpp (id: 743)
LSP: cpptools/getDocumentSymbols: file:///users/user/Downloads/test.cpp (id: 744)
  tag parsing file: C:\Users\user\Downloads\test.cpp
LSP: textDocument/didChange: file:///users/user/Downloads/test.cpp
LSP: cpptools/textEditorSelectionChange
LSP: textDocument/completion: file:///users/user/Downloads/test.cpp (id: 745)
auto_complete::handle_completion: file:///users/user/Downloads/test.cpp (1:4)
LSP: cpptools/getFoldingRanges: file:///users/user/Downloads/test.cpp (id: 746)
LSP: cpptools/getCodeActions: file:///users/user/Downloads/test.cpp (id: 747)
LSP: cpptools/getDocumentSymbols: file:///users/user/Downloads/test.cpp (id: 748)
  tag parsing file: C:\Users\user\Downloads\test.cpp
LSP: cpptools/getInlayHints: file:///users/user/Downloads/test.cpp (id: 749)
Checking for syntax errors: C:\Users\user\Downloads\test.cpp
Queueing IntelliSense update for files in translation unit of: C:\Users\user\Downloads\test.cpp
sending 1 changes to server
sending 1 changes to server
sending 1 changes to server
Error squiggle count: 0
LSP: Request canceled by server (cpptools/getSemanticTokens, id: 743)
Update IntelliSense time (sec): 0.253
LSP: cpptools/getFoldingRanges: file:///users/user/Downloads/test.cpp (id: 750)
LSP: cpptools/getSemanticTokens: file:///users/user/Downloads/test.cpp (id: 751)
```
```


### Other Extensions

_No response_

### Additional context

_No response_

**Comment by michelleangela** [comment]
Fixed in pre-release 1.16.1 https://github.com/microsoft/vscode-cpptools/releases/tag/v1.16.1.

**Comment by michelleangela** [comment]
Fix is now available in [release version 1.16.2](https://github.com/microsoft/vscode-cpptools/releases/tag/v1.16.2).
